bin/spack_pwsh.ps1

```diff
@@ -3,57 +3,5 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-Push-Location $PSScriptRoot/..
-$Env:SPACK_ROOT = $PWD.Path
-Push-Location $PWD/..
-$Env:spackinstdir = $PWD.Path
-Pop-Location
-
-Set-Variable -Name python_pf_ver -Value (Get-Command -Name python -ErrorAction SilentlyContinue).Path
-
-# If python_pf_ver is not defined, we cannot find Python on the Path
-# We next look for Spack vendored copys
-if ($null -eq $python_pf_ver)
-{
-    $python_pf_ver_list = Resolve-Path -Path "$PWD\Python*"
-    if ($python_pf_ver_list.Length -gt 0)
-    {
-        $py_path = $python_pf_ver_list[$python_pf_ver_list.Length-1].Path
-        $py_exe = "$py_path\python.exe"
-    }
-    else {
-        Write-Error -Message "Python was not found on system"
-        Write-Output "Please install Python or add Python to the PATH"
-    }
-}
-else{
-    Set-Variable -Name py_exe -Value $python_pf_ver
-}
-
-if (!$null -eq $py_path)
-{
-    $Env:Path = "$py_path;$Env:Path"
-}
-
-if (!$null -eq $py_exe)
-{
-    Invoke-Expression "$py_exe $Env:SPACK_ROOT\bin\haspywin.py"
-    Invoke-Expression "$py_exe $Env:SPACK_ROOT\bin\spack external find python" | Out-Null
-}
-
-$Env:Path = "$Env:SPACK_ROOT\bin;$Env:Path"
-$Env:EDITOR = "notepad"
-
-
-Write-Output "*****************************************************************"
-Write-Output "**************** Spack Package Manager **************************"
-Write-Output "*****************************************************************"
-
-$command = 'function global:prompt
-{
-    $pth = $(Convert-Path $(Get-Location)) | Split-Path -leaf
-    "[spack] PS $pth>"
-}'
-$bytesc = [System.Text.Encoding]::Unicode.GetBytes($command)
-$encoded_command = [Convert]::ToBase64String($bytesc)
-powershell.exe -NoLogo -encodedCommand $encoded_command -NoExit
+$Env:SPACK_PS1_PATH="$PSScriptRoot\..\share\spack\setup-env.ps1"
+& (Get-Process -Id $pid).Path -NoExit { . $Env:SPACK_PS1_PATH }
```

Comment:
On Windows with the spack_cmd.bat, invoking this script, as an executable or otherwise, should drop the user into the spack root, the same behavior as invoking vcvars on Windows. The hope with the powershell script was to create a powershell analog.
I do however, see the significant use case for being able to invoke the setup script from an existing PowerShell env and remain in the current directory.
I think a fair alternative would be to allow this bin powershell script to be invoked directly and drop the user into the spack root directly, and leave the setup script as is.

Reply:
I'm not sure I parsed your comment correctly, the bin command runs an instance of powershell with the spack environment included in it.  Are you saying it should change directory to the spack root as well?

Reply:
Sorry that my comment was unclear. That is what I'm suggesting, that the powershell script in the bin directory set the directory to the spack root.